File: back-end/src/main/java/es/ull/project/application/usecase/workflow/WorkFlowExportUseCase.java
package es.ull.project.application.usecase.workflow;

import es.ull.project.domain.valueobject.ExportFile;
import es.ull.project.domain.entity.WorkFlow;

import java.util.Optional;


public interface WorkFlowExportUseCase {

    /**
     * Export a workflow entity as a JSON file and return its binary representation and name.
     *
     * @param workflow the workflow to export
     * @return an ExportFile object with filename and byte data
     */
    Optional<ExportFile> exportWorkFlow(WorkFlow workflow);
}

File: front-end/src/application/usecase/workflow/workflow-export-use-case.ts
import { Either } from '@ull-tfg/ull-tfg-typescript';
import type { DataError } from '@ull-tfg/ull-tfg-typescript';
import { ExportFile } from '@/domain/value-object/export-file';
import { UllUUID } from '@ull-tfg/ull-tfg-typescript';

export interface WorkFlowExportUseCase {
    
    /**
     * Export a workflow by its ID.
     * 
     * @param id - Id
     * @returns ExportFile that represents the exported workflow.
     */
    exportWorkFlow(id: UllUUID): Promise<Either<DataError, ExportFile>>;
}

